### PR TITLE
LOOP-3830 PumpManager sync updates

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3830

### CODE REVIEW

**Review by nhamming — APPROVED** [review]
LGTM

File: DashKit/DashPumpManager.swift [hunk 1]
```diff
@@ -680,26 +684,19 @@ open class DashPumpManager: PumpManager {
...
                     self.log.default("Recommending Loop")
                     self.finalizeAndStoreDoses()
-                    self.pumpDelegate.notify({ (delegate) in
-                        completion?()
-                        delegate?.pumpManagerRecommendsLoop(self)
-                    })
+                    completion?(self.lastSync)
                 case .failure(let error):
                     self.log.default("Not recommending Loop because pump data is stale: %@", String(describing: error))
+                    completion?(self.lastSync)
```
Comment (rickpasetto) [inline DashKit/DashPumpManager.swift:690]: [nit] should this `completion?(self.lastSync)` call be inside the `notify` block below?

**Review by rickpasetto — APPROVED** [review]
LGTM

File: DashKit/DashPumpManager.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift:690]: The completion handler doesn't have any specific constraints on what thread it calls on; the notify block is a mechanism for calling methods on the delegate.

**Pull request merged**